src/basic/in-addr-util.h

```diff
@@ -65,7 +65,9 @@ static inline size_t FAMILY_ADDRESS_SIZE(int family) {
         return family == AF_INET6 ? 16 : 4;
 }
 
-#define IN_ADDR_NULL ((union in_addr_union) {})
+/* Workaround for clang, explicitly specify the maximum-size element here.
+ * See also oss-fuzz#11344. */
+#define IN_ADDR_NULL ((union in_addr_union) { .in6 = {} })
```

Comment:
one could argue that this is actually in line with the C standard... gcc allows initialization with empty structs, but c99 requires at least one field to be initialized. Hence our macro relies on the gcc extension...

i figure it's OK merging this hence, but I figure we should report this upstream to oss-fuzz? because they are apparently supporting the gcc extension here but misunderstanding it?

Reply:
Hmm. I am not sure... I am not familiar with what is C standard or gcc extenstion... @keszybz @evverx What do you think about that?

Reply:
I think the workaround is OK. Using a macro instead of the open-coded structure initialized is probably more readable anyway. OTOH, we use such initializations in many many other places, so if we have problems here, we might encounter them in other places too. But let's face that question if we encounter problems in other places.